### ISSUE TITLE
`async = true` does not call `hook` ?

### ISSUE THREAD

**Issue by Lcchy** [actionshrimp/direnv.nvim#2]
Hi there, thanks a lot for this great plugin, its very useful to me as i am trying to finish setting up neovim with nix devShells.

I want to use `async = true` as my .envrc can take some time to load, but it seems that it does not call the `hook` function and hence does not load the .envrc? 
Shouldnt it call `hook` in here?
https://github.com/actionshrimp/direnv.nvim/blob/800024def80c91fcbafc4e01bbdf0a9841e06803/lua/direnv-nvim.lua#L104)

I may have overseen something as I am rather new to neovim.

Thanks for any help and advice, I will also try to test my suspicion and will report back.

**Comment by actionshrimp** [comment]
Thanks for reporting! I don't tend to use `async` at the minute after I found out about [`nix_direnv_manual_reload`](https://github.com/nix-community/nix-direnv?tab=readme-ov-file#manual-reload-of-the-nix-environment) which makes the regular synchronous setup a lot snappier - I overlooked this in a rather over-eager refactor.

Hopefully it should work a bit better for you now but let me know if you run into any other issues with it.
